enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,227 @@
+  - "@deads2k"
+creation-date: 2022-07-08
+last-updated: 2022-07-26
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "https://issues.redhat.com/browse/NE-555"
+see-also:
+replaces:
+superseded-by:
+---
+
+# LoadBalancer Allowed Source Ranges
+
+## Summary
+
+This enhancement extends the IngressController API to allow the user to specify a 
+list of IP address ranges for an IngressController to which access to the LoadBalancer Service 
+should be restricted when `endpointPublishingStrategy` is `LoadBalancerService`. 
+By default, all source addresses are allowed (which is equivalent to
+specifying "0.0.0.0/0"). Currently, the only way for the customers to limit this is by 
+setting `.Spec.LoadBalancerSourceRanges` field for the LoadBalancer Service 
+in front of the router in cloud environments. This is not desired as we prefer
+the customers to use the APIs and to not directly modify the managed resources.
+
+## Motivation
+
+Customer security teams are concerned that the public ELB (created in 
+the installation process by default) is allowing all inbound traffic on 0.0.0.0 
+on ports 80 and 443, so they wish to restrict it. 
+
+Customers have been setting `.Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service themselves, something that we did not plan to support but also did not 
+actively reject. We recognize that users consider it now to be an existing feature 
+while we prefer they use APIs and not directly modify managed resources.
+
+The main motivation of this enhancement is to allow the customers to enable 
+and configure `.Spec.LoadBalancerSourceRanges` through a new API.
+
+### User Stories
+
+#### As a cluster admin, when configuring an IngressController, I want to limit access to the load balancer to a specified list of IP address ranges on the cloud provider for this ingress
+
+The cluster admin has the option to modify the IngressController to specify 
+a list of IP address ranges for an IngressController to which access to 
+the load balancer should be restricted. This is available as a Day-2 operation 
+to the cluster admin.
+
+### Goals
+
+Enable the configuration of a range for a load balancer via the `IngressController.Spec`, 
+specifically via the `LoadBalancerStrategy`, with a new parameter `AllowedSourceRanges`, 
+which represents a list of IP address ranges. This is then used by the operator to set
+`.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service in front of the router 
+in cloud environments.
+
+If `.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service is already set by the 
+customer, its value is used to set `AllowedSourceRanges` if and only if `AllowedSourceRanges`
+was not initially set. Otherwise, `.Spec.LoadBalancerSourceRanges` is overwritten by `AllowedSourceRanges`.
+
+### Non-Goals
+
+N/A
```

Comment:
To reduce risk of confusion, let's note the following:  This enhancement does not change the behavior or API for LoadBalancer-type services in general (that is, services that aren't managed by the ingress operator).

Reply:
Added.